packages/superset-ui-core/src/chart/types/Base.ts

```diff
@@ -7,13 +7,24 @@ export enum Behavior {
   NATIVE_FILTER = 'NATIVE_FILTER',
 }
 
-export type SetExtraFormDataHook = {
+export type SetDataMaskHook = {
   ({
-    extraFormData,
-    currentState: { value },
+    native,
+    cross,
+    own,
   }: {
-    extraFormData: ExtraFormData;
-    currentState: { value: any; [key: string]: any };
+    native?: {
+      extraFormData: ExtraFormData;
+      currentState: { value: any; [key: string]: any };
+    };
+    cross?: {
+      extraFormData: ExtraFormData;
+      currentState: { [key: string]: any };
+    };
+    own?: {
+      extraFormData: ExtraFormData;
+      currentState: { [key: string]: any };
+    };
```

Comment:
When seeing this like this, I think new developers will be confused by what `native`, `cross` and `own` mean. Not sure I have better proposals, but at the risk of being misleading if the scope of this functionality is expanded more, I think `nativeFilters`, `crossFilters` and `ownFilters` could be slightly more descriptive. Alternatively we might want to add a comment per prop to help understand what the context is.